controllers/metal3machine_controller_test.go

```diff
@@ -85,19 +84,20 @@ func setReconcileNormalExpectations(ctrl *gomock.Controller,
 		if tc.AssociateFails {
 			m.EXPECT().Associate(context.TODO()).Return(errors.New("Failed"))
 			m.EXPECT().GetBaremetalHostID(context.TODO()).MaxTimes(0)
-			m.EXPECT().Update(context.TODO()).MaxTimes(0)
 			return m
 		} else {
 			m.EXPECT().Associate(context.TODO()).Return(nil)
+			m.EXPECT().Update(context.TODO()).MaxTimes(0)
```

Comment:
In both cases Update should not be called, so you can add `m.EXPECT().Update(context.TODO()).MaxTimes(0)` outside of the if/else block

Reply:
Fixed